xlators/mgmt/glusterd/src/glusterd-mgmt.c

```diff
@@ -1002,6 +995,13 @@ glusterd_mgmt_v3_pre_validate(glusterd_op_t op, dict_t *req_dict,
         }
     }
 
+    rsp_dict = dict_new();
+    if (!rsp_dict) {
+        gf_msg(this->name, GF_LOG_ERROR, 0, GD_MSG_DICT_CREATE_FAIL,
+               "Failed to create response dictionary");
+        goto out;
+    }
```

Comment:
Why the shift? Just adding a `dict_unref` below in out section won't do?

Reply:
It will do, but if the quorum validation @990 itself fails, there is no need to allocate the dict at all. Hence the shift.